dao/GlossaryDAO.php

```diff
@@ -0,0 +1,109 @@
+    function getDomain($domain_id = 0, $limit = 0, $offset = 0) {   
+        $arr = array();
+        $query = "SELECT *
+                    FROM `{$this->p}glossary_domain` `domain`";
+                    
+        if ($domain_id > 0) {
+            $query .= "WHERE `domain`.`id`=:domain_id";
+            $arr = array(':domain_id' => $domain_id);
+        }
+
+        return $this->PDOX->allRowsDie($query, $arr);
+    }
+    function getAlphabetTerms($link_id, $alphabet) {    
+         $query = "SELECT `id`, `domain_id`, `term`, `description`  
+                    FROM `{$this->p}glossary_term` 
+                    WHERE  `term` LIKE :alphabetPrefix";     
+                    $arr = array(   ':alphabetPrefix' => $alphabet . '%'    );     return $this->PDOX->allRowsDie($query, $arr); }
+    function getTermsForDomain($domain_id = 0, $limit = 0, $offset = 0) {   
+        $arr = array();
+        $query = "SELECT `def`.`id`, `domain`.`name` as `domain_name`,`def`.`term`,`def`.`description` 
+                    FROM `{$this->p}glossary_term` `def` 
+                    LEFT JOIN `{$this->p}glossary_domain` `domain` on `domain`.`id` = `def`.`domain_id`";
+
+        if ($domain_id > 0) {
+            $query .= "WHERE `domain_id`=:domain_id AND `deleted` = 0";
+            $arr = array(':domain_id' => $domain_id);
+        } else {
+            $query .= "WHERE `deleted` = 0";
+        }
+        // if limit > 0 add to query
+        if ($limit > 0) {
+            $query .= " limit ". $limit;
+        }
+        // same for offset
+        return $this->PDOX->allRowsDie($query, $arr);
+    }
+    
+    function getTerms($id, $domain_id, $term , $description ) {   
+        $arr = array();
+        $query = "SELECT `id`,`domain_id`,`term`,`description` 
+                    FROM `{$this->p}glossary_term`where link_id = :linkId and site_id = :siteId;";
+         return $this->PDOX->allRowsDie($query, $arr);
+    }
+       
+    function getAllTerms() {
+        $result = $this->PDOX->allRowsDie("SELECT `term`.`id`,`term`.`domain_id`,`term`.`term`,`term`.`description`,`domain`.`name` AS `domain`
+                    FROM `{$this->p}glossary_term` AS `term`
+                    LEFT JOIN `glossary_domain` AS `domain` ON `term`.`domain_id` = `domain`.`id`
+                    WHERE `deleted` = 0 
+                    order by `term`.`term` asc");
+
+        return $result;
+    }
+
+    // function getTermsForLanguage($selectedLanguage) {
+    //     $query = "SELECT `term `
+    //                 FROM `{$this->p}glossary_term_translation_text`
+    //                 LEFT JOIN `{$this->p}glossary_term_translation_text`,`glossary_term_translation` on `glossary_term_translation`.`id` = `glossary_term_translation_text`.`translation_id`
+    //                 WHERE `language_id` = ?";
+
+    //     return $this->PDOX->allRowsDie($query);
+    // }
+
+    function getAllLanguages() {   
+        $query = "SELECT `id`,`language`
```

Comment:
reduce - single line `return $this->PDOX->allRowsDie("SELECT `id`,`language` FROM `{$this->p}glossary_language`");`